### PR TITLE
fix: handle learning curve edge cases [DET-4832]

### PR DESCRIPTION
**Pull request by hkang1**
## Description

Better handle all the different edge cases for HP-viz page.

In progress experiment that does not have any data to plot yet:
Screen Shot 2021-01-20 at 5 28 43 PM

Terminal experiment (errored, killed, canceled) experiments that do not have any data to show:
Screen Shot 2021-01-20 at 5 32 50 PM

Single trial or PBT experiments alert message (we do not support HP-viz for those two types):
Screen Shot 2021-01-20 at 5 33 18 PM

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/react/src/ioTypes.ts [hunk 1]
```diff
@@ -268,6 +271,7 @@ export const ioExperimentConfig = io.type({
   resources: ioExpResources,
   searcher: io.type({
     metric: io.string,
+    name: io.keyof(experimentSearcherName),
```
Comment (hamidzr) [inline webui/react/src/ioTypes.ts:274]: test suggestion: this should be running through our experiment config fixtures set. could you maybe add some configs that have names that are missing from our set?
the set currently has these
```
        "name": "adaptive",
        "name": "adaptive_asha",
        "name": "adaptive_simple",
        "name": "single",
```

File: webui/react/src/pages/ExperimentDetails/ExperimentVisualization.tsx [hunk 2]
```diff
@@ -59,6 +69,7 @@ const ExperimentVisualization: React.FC<Props> = ({
   /* eslint-disable-next-line */
   const [ batches, setBatches ] = useState<number[]>([]);
   const [ hasLoaded, setHasLoaded ] = useState(false);
+  const [ pageError, setPageError ] = useState<PageError>();
```
Comment (hamidzr) [inline webui/react/src/pages/ExperimentDetails/ExperimentVisualization.tsx:72]: praise: 👍🏼

Comment (hamidzr) [reply webui/react/src/pages/ExperimentDetails/ExperimentVisualization.tsx:72]: maybe we'd have a global hook and overlay ui for these use cases

File: webui/react/src/types.ts [hunk 3]
```diff
@@ -197,6 +197,16 @@ interface ExperimentHyperParam {
 
 export type ExperimentHyperParams = Record<string, ExperimentHyperParam>;
 
+export enum ExperimentSearcherName {
+  AdaptiveAdvanced = 'adaptive',
```
Comment (hamidzr) [inline webui/react/src/types.ts:201]: @katport does this look good to you? are these all the experiment searcher names? it looks to be all of what https://docs.determined.ai/latest/reference/experiment-config.html#experiment-configuration-searcher talks about

File: webui/react/src/pages/ExperimentDetails/ExperimentVisualization.tsx [hunk 4]
```diff
@@ -109,7 +119,10 @@ const ExperimentVisualization: React.FC<Props> = ({
         setValidationMetrics(newValidationMetrics);
         if (event.searcherMetric) setSearcherMetric(event.searcherMetric);
       },
-    );
+    ).catch(() => {
+      setHasLoaded(true);
```
Comment (hamidzr) [inline webui/react/src/pages/ExperimentDetails/ExperimentVisualization.tsx:123]: I think we could use a type like the one we defined in useRest for async request which has: `error` `hasLoaded` `data` or something similar here and many other places where we make api calls

**Review by hamidzr — APPROVED** [review]
LGTM

File: webui/react/src/ioTypes.ts (hunk 1 above)
Comment (hkang1) [reply webui/react/src/ioTypes.ts:274]: I looked as the file `src/fixtures/responses/experiment-details/set-a.json` and it has all 4 name types, so we should be good right?

Comment (hkang1) [reply webui/react/src/ioTypes.ts:274]: nm it's missing `grid`, `pbt`, and `random`

File: webui/react/src/pages/ExperimentDetails/ExperimentVisualization.tsx (hunk 4 above)
Comment (hkang1) [reply webui/react/src/pages/ExperimentDetails/ExperimentVisualization.tsx:123]: We might be primed for something like this for the new api / consumeStream!

File: webui/react/src/types.ts (hunk 3 above)
Comment (hkang1) [reply webui/react/src/types.ts:201]: I got these directly from the docs, but I couldn't actually find any examples in our repot for `pbt`, `random` or `grid`...

File: webui/react/src/ioTypes.ts (hunk 1 above)
Comment (hkang1) [reply webui/react/src/ioTypes.ts:274]: I'm working with @katport to figure out fixtures for these 3, but I'll add them in as a separate task.

Comment (hkang1) [reply webui/react/src/ioTypes.ts:274]: I've added 3 exp config for `pbt`, `random` and `grid`

File: webui/react/src/pages/ExperimentDetails/ExperimentVisualization.tsx (hunk 2 above)
Comment (hkang1) [reply webui/react/src/pages/ExperimentDetails/ExperimentVisualization.tsx:72]: agreed. leave for a different PR?

**Pull request merged**